### PR TITLE
More kustomization for demo and easy install

### PR DESCRIPTION
**Pull request by costinm**
- add an 'istio-testing' base that will install the demo in istio-testing namespace
- add a kustomize base for crds and ingress ( still needs makefile changes to generate the default ingress)
For 1.2 we can probably allow at least additional gateways to be deployed 'the easy way' via kubectl apply and kustomize.

### CODE REVIEW

File: test/demo/istio-testing/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Installs istio-demo profile in istio-testing environment
+# TODO: override as much as possible
```
Comment (howardjohn) [inline test/demo/istio-testing/kustomization.yaml]: What are we overriding? To get the demo values (resource settings, enabled, etc)?

if so shouldnt we just generate those using helm values?

File: gateways/istio-ingress/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - istio-ingress.yaml
```
Comment (howardjohn) [inline gateways/istio-ingress/kustomization.yaml]: This doesn't seem to be a file that exists?

File: test/demo/istio-testing/kustomization.yaml (hunk 1 above)
Comment (costinm) [reply test/demo/istio-testing/kustomization.yaml]: No, we do have the right demo values in test/demo - this is to test how much we can customize the demo with only kustomize.

The 'base' demo is using istio-system - and if we're lucky we may be able to use it as an in-place upgrade from 1.1 ( seems doable - not sure if worth it ).

This is a profile that installs in istio-testing, and will test the limits of kustomization ( == override as much as we can )

File: gateways/istio-ingress/kustomization.yaml (hunk 2 above)
Comment (costinm) [reply gateways/istio-ingress/kustomization.yaml]: fixed - I still don't know if checking in generated files is right, but golang gen files and vendor are doing it so we may be ok for 1.2

**Comment by costinm** [comment]
The main challenge remains citadel in istio-system - still trying to find how to properly configure it

File: gateways/istio-ingress/kustomization.yaml (hunk 2 above)
Comment (howardjohn) [reply gateways/istio-ingress/kustomization.yaml]: I think its ok, as long as its deterministic. There was some stuff different when I ran it vs yours - some stuff has `costin` in it at the very least and I think some may not have consistent ordering.

**Pull request merged**